File: arb-www/src/schema/sportradar/nhl/teams.ts
import { BaseDTO } from '../../BaseDTO';

/**
 * NHL Team Profile information
 */
export class NHLTeamProfile extends BaseDTO {
  TeamID!: number;
  Key!: string;
  Active!: boolean;
  City!: string;
  Name!: string;
  StadiumID!: number;
  Conference!: string;
  Division!: string;
  PrimaryColor!: string;
  SecondaryColor!: string;
  TertiaryColor!: string;
  QuaternaryColor?: string;
  WikipediaLogoUrl!: string;
  WikipediaWordMarkUrl?: string;
  GlobalTeamID!: number;
  HeadCoach!: string;
}

/**
 * NHL Team Profiles response
 */
export class NHLTeamProfilesResponse extends BaseDTO {
  data!: NHLTeamProfile[];
  league!: string;
}

File: arb-rs/src/schema/sportradar/nhl/teams.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NHLTeamProfile {
    #[serde(rename = "TeamID")]
    pub team_id: i32,

    #[serde(rename = "Key")]
    pub key: String,

    #[serde(rename = "Active")]
    pub active: bool,

    #[serde(rename = "City")]
    pub city: String,

    #[serde(rename = "Name")]
    pub name: String,

    #[serde(rename = "StadiumID")]
    pub stadium_id: i32,

    #[serde(rename = "Conference")]
    pub conference: String,

    #[serde(rename = "Division")]
    pub division: String,

    #[serde(rename = "PrimaryColor")]
    pub primary_color: String,

    #[serde(rename = "SecondaryColor")]
    pub secondary_color: String,

    #[serde(rename = "TertiaryColor")]
    pub tertiary_color: String,

    #[serde(rename = "QuaternaryColor")]
    pub quaternary_color: Option<String>,

    #[serde(rename = "WikipediaLogoUrl")]
    pub wikipedia_logo_url: String,

    #[serde(rename = "WikipediaWordMarkUrl")]
    pub wikipedia_word_mark_url: Option<String>,

    #[serde(rename = "GlobalTeamID")]
    pub global_team_id: i32,

    #[serde(rename = "HeadCoach")]
    pub head_coach: String,
}
